generators/server/index.js

```diff
@@ -278,6 +282,46 @@ module.exports = JhipsterServerGenerator.extend({
                 },
                 {
                     when: function (response) {
+                        return applicationType === 'gateway' || applicationType == 'microservice';
+                    },
+                    type: 'list',
+                    name: 'authenticationType',
+                    message: function (response) {
+                        return getNumberedQuestion('Which *type* of authentication would you like to use?', currentQuestion, totalQuestions, function (current) {
+                            currentQuestion = current;
+                        }, applicationType === 'monolith');
```

Comment:
condition should be `applicationType === 'gateway' || applicationType === 'microservice'`